### PR TITLE
IGNITE-16083 .NET: Java service invocation with thin client

### PR DESCRIPTION
**Pull request by nizhikov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by nizhikov** [comment]
Hello @ptupitsyn 

Can you, please, take a look at my changes?

File: modules/platforms/dotnet/Apache.Ignite.Core/Client/Services/IServicesClient.cs [hunk 1]
```diff
@@ -51,5 +51,12 @@ public interface IServicesClient
         /// </summary>
         /// <returns>Instance with server-side binary mode enabled.</returns>
         IServicesClient WithServerKeepBinary();
+
+        /// <summary>
+        /// Returns an instance for obtaining .Net services.
+        /// Service method serialization and deserialization will be .Net specific.
+        /// </summary>
+        /// <returns>Instance for obtaining platform services.</returns>
+        IServicesClient ForDotNetService();
```
Comment (ptupitsyn) [inline modules/platforms/dotnet/Apache.Ignite.Core/Client/Services/IServicesClient.cs:60]: I think we should use the same approach as in thick service API and detect service type automatically. Let's not shift this responsibility to the user.

https://github.com/apache/ignite/blob/cca909ca92965aa12dca5139ec2803df4ffbc4d4/modules/platforms/dotnet/Apache.Ignite.Core/Impl/Services/Services.cs#L392

**Review by ptupitsyn — CHANGES_REQUESTED** [review]

File: modules/platforms/dotnet/Apache.Ignite.Core/Client/Services/IServicesClient.cs (hunk 1 above)
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite.Core/Client/Services/IServicesClient.cs:60]: We may want to implement https://issues.apache.org/jira/browse/IGNITE-13359 first for that.

**Review by ptupitsyn — APPROVED** [review]
Looks good to me.

**Pull request merged**